source/slang/slang-check-overload.cpp

```diff
@@ -1432,6 +1447,18 @@ namespace Slang
             auto type = DeclRefType::create(m_astBuilder, genericTypeParamDeclRef);
             AddTypeOverloadCandidates(type, context);
         }
+        else if( auto localDeclRef = item.declRef.as<ParamDecl>() )
+        {
+            // We could probably be broader than just parameters here
+            // eventually.
+            // Limit it for now though to make the specialization easier
+            ensureDecl(localDeclRef, DeclCheckState::CanUseFuncSignature);
+            const auto type = localDeclRef.getDecl()->getType();
+            if(const auto funType = as<FuncType>(type))
+                AddFuncExprOverloadCandidate(funType, context, context.originalExpr->functionExpr);
+            else
+                SLANG_UNREACHABLE("Trying to find overload for non-function typed parameter");
```

Comment:
This will cause intellisense to completely stop working for the current file since it will throw an exception. Consider more graceful handling of errors like this in front end, issueing an unimplemmted diagnostic maybe the best way.

Reply:
I think the intention here was that it could never happen. would `SLANG_ASSERT(funType)` be clearer?

Reply:
Hmm, but if assert turns into an assume in release build, it may cause problems.

Reply:
ok, seems like it was unnecessary anyway as we handle this function not adding any overload candidates gracefully.

I added an explicit return on the else branchs to make this control flow clear.